src/make-hid.sh

```diff
@@ -1,25 +1,74 @@
-#!/bin/sh
+#!/bin/bash
+
 MAC=$(echo "$1" | tr a-f A-F)
-DEV=hci0
-CTRL=$(tr a-f A-F </sys/class/bluetooth/$DEV/address)
+HCIDEV=hci0
+PASSEDDEV="$2"
 
 function die() {
   echo "FATAL: $@" 1>&2
   exit 1
 }
 
-[ -z "$CTRL" ] || [ -z "$MAC" ] && die "Call as: $0 keyboard-mac"
+declare -A ahci
+
+ahci=()
+pipe=/tmp/hcitool$$
+trap "rm -f $pipe" EXIT
+
+if [[ ! -p $pipe ]]; then
+    echo  mkfifo $pipe
+fi
+
+hcitool dev > $pipe 2> /dev/null
+while read -t 1 line
+do
+    case $line in
+	hci*)
+	    strs=($line)
+	    echo "	hci[${strs[0]##hci}]: " ${strs[1]}
+	    ahci+=([${strs[0]}]="${strs[1]}")
+	    # echo "hci[${strs[0]}]: "${hci[${strs[0]}]}
+	    ;;
+	*)
+	    echo "$line"
+	    ;;
+    esac
+    
+done < $pipe
+
+if hcitool dev | grep "hci" | wc -l > 1 ; then
+    if [ "$PASSEDDEV" == "" ]; then
+	echo "There are multiple HCI devices."
```

Comment:
please fix the indenting everywhere :)

Reply:
Fixed in commit 74a2406d2b88448ad483194856365e5755c0f6e4